### PR TITLE
fix: Pasting location coordinates into field of type `GeoPoint` does not work in data browser

### LINKED ISSUE #2462: Allow to paste coordinates tuple in data browser
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

### Current Limitation

While it's possible to select a geo point field in the data browser and copy its content tuple (long, lat), it's not possible to paste the tuple. This makes it difficult to copy a location from one field and paste it into another field as long and lat need to be pasted individually in two separate text fields.

https://github.com/parse-community/parse-dashboard/assets/5673677/1d451abc-63c8-4b8e-9538-6fa844ea6c3d

### Feature / Enhancement Description

Allow to paste the copies geo point in a single step into a geo point field.

### Example Use Case

1. In data browser select a geo point field.
2. Copy the field value Crtl+C; adds  `(12.345, 12.345)` to the pasteboard.
3. Select another geo point field, double click and use Crtl+V to paste `(12.345, 12.345)`.

### Alternatives / Workarounds
Copy and paste long and lat separate to a new geo field.

### PR DESCRIPTION
**Pull request by patelmilanun**
- add formats like below to be paste-able in geo point field
    - (x, y)
    - (x,y)
    - x, y
    - x,y
  
### New Pull Request Checklist
- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

### Issue Description
There was a paste functionality for GeoPoint that was accepting the x,y format already. But as when we are copying the field we are getting the format like (x, y) which makes the existing functionality outdated.

Closes: #2462 

### Approach
I basically added other formats including (x, y) that will get accepted when pasting.

### TODOs before merging
NA

### CODE REVIEW

**Comment by mtrezza** [comment]
> There was a paste functionality for GeoPoint that was accepting the x,y format already. But as when we are copying the field we are getting the format like (x, y) which makes the existing functionality outdated.

Who would have thought. So this is actually a bug fix I suppose.

**Comment by patelmilanun** [comment]
Yes, we can say that. It now accepts total of 4 format.

File: src/components/GeoPointEditor/GeoPointEditor.react.js [hunk 1]
```diff
@@ -106,26 +106,32 @@ export default class GeoPointEditor extends React.Component {
...
-            }
-
-            if (validateNumeric(values[1])) {
-              this.setState({ latitude: values[0] });
-              this.setState({ longitude: values[1] });
-              this.longitudeRef.current.focus();
-              return;
-            }
+        // This regex will match this form of input: (x, y) or (x,y) or x, y or x,y
+        const regex = /\((-?\d+\.?\d*),\s*(-?\d+\.?\d*)\)|(-?\d+\.?\d*),\s*(-?\d+\.?\d*)/;
```
Comment (mtrezza) [inline src/components/GeoPointEditor/GeoPointEditor.react.js]: The regex seems too complex while being quite strict. This is easier to read and extend in the future, while being more tolerant:

`[\[("' ]?(?<x>[0-9.]+)["' ]?,["' ]?(?<y>[0-9.]+)["' )\]]?`

It accepts:

```
1.0,1.0
1.0, 1.0
(1.0,1.0)
("1.0","2.0")
['1.0','1.0']
```

I'm not sure about different locales though, how should we support that? If a user's locale uses a comma as decimal marker, then the input could be `("1,0", "1,0"`). But maybe we can ignore that for now sine the dashboard is not i18n'ed anyway.

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: src/components/GeoPointEditor/GeoPointEditor.react.js (hunk 1 above)
Comment (patelmilanun) [reply src/components/GeoPointEditor/GeoPointEditor.react.js]: ok i'll go with your suggestion

Comment (mtrezza) [reply src/components/GeoPointEditor/GeoPointEditor.react.js]: Great, just a note, the regex contains named capture groups, which should make it easier to extract the `x` and `y` values.

```
const regex = /[\[("' ]?(?<x>[0-9.]+)["' ]?,["' ]?(?<y>[0-9.]+)["' )\]]?/;
const input = '[3.14, 2.718]';
const match = regex.exec(input);
const x = match.groups.x;
const y = match.groups.y;
```

Comment (patelmilanun) [reply src/components/GeoPointEditor/GeoPointEditor.react.js]: it does not accept ("1.0","2.0") , ['1.0','1.0'] basically square brackets and quote with parentheses.

Comment (mtrezza) [reply src/components/GeoPointEditor/GeoPointEditor.react.js:110]: let me check...

Comment (mtrezza) [reply src/components/GeoPointEditor/GeoPointEditor.react.js]: The regex works, here's the test:

```
const values = [
  '1.2,3.4',
  '1.2, 3.4',
  '(1.2,3.4)',
  '("1.2","3.4")',
  `['1.2','3.4']`,
];

const regex = /[\[("' ]?(?<x>[0-9.]+)["' ]?,["' ]?(?<y>[0-9.]+)["' )\]]?/;

for (const value of values) {
  const match = regex.exec(value);
  const x = match.groups.x;
  const y = match.groups.y;
  const correct = parseFloat(x) === 1.2 && parseFloat(y) === 3.4;
  console.log('correct: ' + correct);
}
```

So it must be another issue. I guess the pasted value is different.

Comment (patelmilanun) [reply src/components/GeoPointEditor/GeoPointEditor.react.js]: sorry i was not considering the fact that it is a named capture groups. Now its working. Thanks.

File: src/components/GeoPointEditor/GeoPointEditor.react.js [hunk 2]
```diff
@@ -106,26 +106,31 @@ export default class GeoPointEditor extends React.Component {
...
-              return;
-            }
-
-            if (validateNumeric(values[1])) {
-              this.setState({ latitude: values[0] });
-              this.setState({ longitude: values[1] });
-              this.longitudeRef.current.focus();
-              return;
-            }
+        // This regex will match this form of input: (x, y) or (x,y) or x, y or x,y and many more
```
Comment (mtrezza) [inline src/components/GeoPointEditor/GeoPointEditor.react.js]:
```suggestion
```

File: src/components/GeoPointEditor/GeoPointEditor.react.js [hunk 3]
```diff
@@ -106,26 +106,30 @@ export default class GeoPointEditor extends React.Component {
...
-            }
-
-            if (validateNumeric(values[1])) {
-              this.setState({ latitude: values[0] });
-              this.setState({ longitude: values[1] });
-              this.longitudeRef.current.focus();
-              return;
-            }
+        const regex =
+          /[[("' ]?(?<x>[0-9.]+)["' ]?,["' ]?(?<y>[0-9.]+)["' )\]]?/;
```
Comment (mtrezza) [inline src/components/GeoPointEditor/GeoPointEditor.react.js:110]: Are you sure this regex works without the escape character?

Comment (mtrezza) [inline src/components/GeoPointEditor/GeoPointEditor.react.js]:
```suggestion
        const regex = /[[("' ]?(?<x>[0-9.]+)["' ]?,["' ]?(?<y>[0-9.]+)["' )\]]?/;
```

**Review by mtrezza — APPROVED** [review]
Looks good, tested in preview and works

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.2.0-alpha.18](https://github.com/parse-community/parse-dashboard/releases/tag/5.2.0-alpha.18)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-beta.1](https://github.com/parse-community/parse-dashboard/releases/tag/5.3.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-alpha.1](https://github.com/parse-community/parse-dashboard/releases/tag/5.3.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0](https://github.com/parse-community/parse-dashboard/releases/tag/5.3.0)

**Pull request merged**